docs/developer-guide/actions/triggers-and-events.rst

```diff
@@ -1,33 +1,79 @@
 
 Lifecycle of a Robusta event
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 1. A pod changes
 2. The API Server notifies Robusta
 3. Robusta checks if any triggers like ``on_pod_update`` are activated by the pod change
 4. If yes, Robusta calls that trigger
-5. The trigger converts data from the APIServer to a concrete event like ``PodChangeEvent``
-6. The ``PodChangeEvent`` is passed to all playbook actions
+5. The trigger converts data from the APIServer to a concrete event like ``PodEvent``
+6. The ``PodEvent`` is passed to all playbook actions
 
+Compatibility of actions and events
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is discussed in :ref:`Trigger-Action Compatibility` from the perspective of a user who configures playbooks in YAML.
+
+Let's look at this from the perspective of a developer who is writing a playbook action in Python.
+
+You declare which events your playbook is compatible with by choosing the first parameter of your playbook's Python function. For example:
+
+.. code-block:: python
+
+    @action
+    def delete_pod(event: PrometheusKubernetesAlert):
+    if not event.get_pod():
+        logging.info("alert is not related to pod")
+        return
+
+    event.get_pod().delete()
+
+The above action takes ``PrometheusKubernetesAlert`` as input and can therefore only be connected to an ``on_prometheus_alert`` trigger as no other trigger generates that event object.
+
+However, this action doesn't actually need a Prometheus alert as input as it does nothing with the alert itself. We can modify this action so it takes a ``PodEvent`` as input.
+
+.. code-block:: python
+
+    @action
+    def delete_pod(event: PodEvent):
+    if not event.get_pod():
+        logging.info("alert is not related to pod")
+        return
+
+    event.get_pod().delete()
+
+``PrometheusKubernetesAlert`` is a subclass of ``PodEvent``, so general object-oriented principles apply here:
+``PrometheusKubernetesAlert`` can be used wherever ``PodEvent`` is expected.
+
+Therefore this action can be triggered with ``on_prometheus_alert`` but it can **also** be triggered on any other PodEvent.
+
+Who is compatible with who
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Here is the Robusta event hierarchy:
 
 .. image:: ../../images/event-inheritance.png
   :align: center
 
-
 ..
+    this is a sphinx comment
     the above image was generated like this by a patched version of inheritance-diagram based on
     https://github.com/sphinx-doc/sphinx/pull/8159
     .. inheritance-diagram2:: robusta.api.ExecutionBaseEvent
         :parts: 1
         :include-subclasses:
 
+
+.. note::
+
+    Note that ``PrometheusKubernetesAlert`` inherits from many Kubernetes event classes, so you can use
+    ``on_prometheus_alert`` with many types of playbook actions. For example, ``on_prometheus_alert`` is
+    compatible with any playbook that accepts ``PodEvent``, ``NodeEvent``, ``DeploymentEvent``, and so on.
```

Comment:
I would add that when doing it, you need to verify that the alert contains the actual resource.
Because, for example, a Node alert might not have a Pod